onnxruntime/core/mlas/lib/platform.cpp

```diff
@@ -449,10 +449,13 @@ Return Value:
 
 #if defined(_WIN32)
     HasDotProductInstructions = (IsProcessorFeaturePresent(PF_ARM_V82_DP_INSTRUCTIONS_AVAILABLE) != 0);
-#else
+#elif !defined(__APPLE__)  // The next few lines result in an EXC_BAD_INSTRUCTION runtime error on a M1 Mac so we
+                           // disable it there.
```

Comment:
The change came from https://github.com/microsoft/onnxruntime/pull/16082

dot product feature is available on almost all apple ARM64 chips, setting this to false for all apple platforms might cause performance problem of the CPU EP on those platforms. The GEMM performance difference would be significant. 
 
Can we try this in apple platforms:
    HasDotProductInstructions = MLAS_CPUIDINFO::GetCPUIDInfo().HasArmNeonDot();

Reply:
Tried `MLAS_CPUIDINFO::GetCPUIDInfo().HasArmNeonDot()` and did not get any error. Will go with that.